apollo-router/src/plugins/rhai.rs

```diff
@@ -355,14 +360,47 @@ mod router_plugin_mod {
     }
 }
 
-/// Plugin which implements Rhai functionality
-#[derive(Default, Clone)]
-pub(crate) struct Rhai {
+struct EngineBlock {
     ast: AST,
     engine: Arc<Engine>,
     scope: Arc<Mutex<Scope<'static>>>,
 }
 
+impl EngineBlock {
+    fn try_new(
```

Comment:
I could add that method, but I wouldn't be using it, so it would just end up with a clippy macro to ignore that. I'd rather not add unused code and it's not much effort to add it if/when it is needed.